lib/api/formtools/grid.js

```diff
@@ -0,0 +1,263 @@
+var linz = require('../../../');
+    async = require('async'),
+    clone = require('clone');
+
+(function  () {
+
+    function getSettings(modelName, cb) {
+
+        if (!modelName || typeof(modelName) !== 'string' || !modelName.length) {
+            throw new Error('modelName is required and must be of type string and cannot be emptied.');
+        }
+
+        var model = linz.get('models')[modelName];
+
+        model.getGrid(function (err, settings) {
+
+            if (err) {
+                return cb(err);
+            }
+
+            return cb(err, clone(settings));
+
+        });
+
+    }
+
+    function getToolbarItems (req, res, modelName, cb) {
+
+        if (!modelName || typeof(modelName) !== 'string' || !modelName.length) {
+            throw new Error('modelName is required and must be of type string and cannot be emptied.');
+        }
+
+        var model = linz.get('models')[modelName];
+
+        async.waterfall([
+
+            function (callback) {
+                return getSettings(modelName, callback);
+            }
+
+        ], function (err, settings) {
+
+            if (err) {
+                return cb(err);
+            }
+
+            if (!settings.toolbarItems.length) {
+                return cb(null);
+            }
+
+            async.each(settings.toolbarItems, function (item, itemDone) {
+
+                if (!item.renderer) {
```

Comment:
Do you think we should throw an error here? A toolbarItem without a renderer is essentially useless. It might help people implementing a toolbarItem to know when they've done something wrong. It will always be an implementation issue rather than a runtime issue too.